### PR TITLE
BUGFIX: Fusion avoid error parser cache to crash if cache is broken

### LINKED ISSUE #4595: Fusion ParserCache crashes when generated value is not a FusionFile
In https://github.com/neos/neos-development-collection/blob/6fcba54e22bdecc4dc70cf689dea2edce2583e8b/Neos.Fusion/Classes/Core/Cache/ParserCache.php#L65 the ParserCache just returns a value from another method that can return a `mixed` value and doesn't check if the value is actually a `FusionFile`. 
That leads to a crash when the generated value is `false` due to some other issue with the composer state or stale file cache.

### PR DESCRIPTION
**Pull request by mhsdesign**
Resolves #4595

It seems a cache's `has` operation doesnt work that reliable thats why we will be using `get` instead and check against false.  

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Review by Sebobo — APPROVED** [review]
LGTM

File: Neos.Fusion/Classes/Core/Cache/ParserCache.php [hunk 1]
```diff
@@ -59,7 +67,18 @@ public function cacheForFusionFile(?string $contextPathAndFilename, \Closure $ge
             $contextPathAndFilename = $this->getAbsolutePathForPackageRessourceUri($contextPathAndFilename);
         }
         $identifier = $this->getCacheIdentifierForFile($contextPathAndFilename);
-        return $this->cacheForIdentifier($identifier, $generateValueToCache);
+        $value = $this->cacheForIdentifier($identifier, $generateValueToCache);
+        if (!$value instanceof FusionFile) {
+            $this->logger->error(sprintf(
```
Comment (mhsdesign) [inline Neos.Fusion/Classes/Core/Cache/ParserCache.php]: maybe we should even use the throwable error storage to empathise on the error?

Comment (kitsunet) [reply Neos.Fusion/Classes/Core/Cache/ParserCache.php]: I wouldn't even categorize this as error. It's a cache, things can happen, it should be able to deal with this.

Comment (mhsdesign) [reply Neos.Fusion/Classes/Core/Cache/ParserCache.php]: okay now that you say this, i agree.

> Is why I almost never use "has" in caches, it is never atomic, you cannot know what happens between has and get.

will update this accordingly. Is there anything we have to fix in flow then for 9.0? Like remove has :D (No just joking as _false_ could be an acceptable value...

Comment (mhsdesign) [reply Neos.Fusion/Classes/Core/Cache/ParserCache.php]: removed the logging ;)

**Review by jonnitto — APPROVED** [review]

**Pull request merged**